File: PSET 6/cash.py
from cs50 import get_float

c = get_float("Change owed:\n")

while c < 0.00:
    c = get_float("Change owed:\n")

rem = round(c * 100)
count = 0

while rem >= 25:
    count += 1
    rem -= 25

while rem >= 10:
    count += 1
    rem -= 10

while rem >= 5:
    count += 1
    rem -= 5

while rem >= 1:
    count += 1
    rem -= 1

print(count)

File: cash.c
// Abstraction and scope

#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    float dollars;
    do
    {
        dollars = get_float("Change owed: ");
    }
    while (dollars < 0);
    printf("%.2f\n", dollars);

    int coins = round(dollars * 100);
    int count = 0;
    int amount_left = coins;

    while (amount_left >= 25)
    {
        count++;
        amount_left -= 25;
    }
    while (amount_left >= 10)
    {
        count++;
        amount_left -= 10;
    }
    while (amount_left >= 5)
    {
        count++;
        amount_left -= 5;
    }
    while (amount_left >= 1)
    {
        count++;
        amount_left -= 1;
    }
    printf("%i\n", count);
}

